src/Compilers/Core/Portable/FileSystem/FileUtilities.cs

```diff
@@ -372,6 +372,21 @@ internal static DateTime GetFileTimeStamp(string fullPath)
             }
         }
 
+        /// <exception cref="IOException"/>
+        internal static long GetFileLength(string fullPath)
+        {
+            Debug.Assert(PathUtilities.IsAbsolute(fullPath));
+            try
+            {
+                var info = new FileInfo(fullPath);
+                return info.Length;
+            }
+            catch (Exception e)
+            {
+                throw new IOException(e.Message, e);
```

Comment:
I'd leave it here and do what we do in the rest of the FileUtilities - which is to turn non-IOExceptions to IOExceptions. It's consistent with the rest of the utils code.